src/perf/bin/drvmain.cpp

```diff
@@ -0,0 +1,604 @@
+        LIST_ENTRY ClientList;
+    ULONG ClientListSize;
+
+} QUIC_DEVICE_EXTENSION;
+
+WDF_DECLARE_CONTEXT_TYPE_WITH_NAME(QUIC_DEVICE_EXTENSION, QuicPerfCtlGetDeviceContext);
+
+typedef struct QUIC_TEST_CLIENT
+{
+    LIST_ENTRY Link;
+    bool TestFailure;
+
+} QUIC_TEST_CLIENT;
+
+WDF_DECLARE_CONTEXT_TYPE_WITH_NAME(QUIC_TEST_CLIENT, QuicPerfCtlGetFileContext);
+
+EVT_WDF_IO_QUEUE_IO_DEVICE_CONTROL QuicPerfCtlEvtIoDeviceControl;
+EVT_WDF_IO_QUEUE_IO_CANCELED_ON_QUEUE QuicPerfCtlEvtIoCanceled;
+
+PAGEDX EVT_WDF_DEVICE_FILE_CREATE QuicPerfCtlEvtFileCreate;
+PAGEDX EVT_WDF_FILE_CLOSE QuicPerfCtlEvtFileClose;
+PAGEDX EVT_WDF_FILE_CLEANUP QuicPerfCtlEvtFileCleanup;
+
+WDFDEVICE QuicPerfCtlDevice = nullptr;
+QUIC_DEVICE_EXTENSION* QuicPerfCtlExtension = nullptr;
+QUIC_TEST_CLIENT* QuicPerfClient = nullptr;
+
+EVT_WDF_DRIVER_UNLOAD QuicPerfDriverUnload;
+
+_No_competing_thread_
+INITCODE
+NTSTATUS
+QuicPerfCtlInitialize(
+    _In_ WDFDRIVER Driver
+    );
+
+_IRQL_requires_max_(PASSIVE_LEVEL)
+VOID
+QuicPerfCtlUninitialize(
+
+    );
+
+void* __cdecl operator new (size_t Size) {
+    return ExAllocatePool2(POOL_FLAG_NON_PAGED, Size, QUIC_PERF_TAG);
+}
+
+void __cdecl operator delete (_In_opt_ void* Mem) {
+    if (Mem != nullptr) {
+        ExFreePoolWithTag(Mem, QUIC_PERF_TAG);
+    }
+}
+
+void __cdecl operator delete (_In_opt_ void* Mem, _In_opt_ size_t) {
+    if (Mem != nullptr) {
+        ExFreePoolWithTag(Mem, QUIC_PERF_TAG);
+    }
+}
+
+void* __cdecl operator new[](size_t Size) {
+    return ExAllocatePool2(POOL_FLAG_NON_PAGED, Size, QUIC_PERF_TAG);
+}
+
+void __cdecl operator delete[](_In_opt_ void* Mem) {
+    if (Mem != nullptr) {
+        ExFreePoolWithTag(Mem, QUIC_PERF_TAG);
+    }
+}
+
+extern "C" _IRQL_requires_max_(PASSIVE_LEVEL) void QuicTraceRundown(void) { }
+
+extern "C"
+INITCODE
+_Function_class_(DRIVER_INITIALIZE)
+_IRQL_requires_same_
+_IRQL_requires_(PASSIVE_LEVEL)
+NTSTATUS
+DriverEntry(
+    _In_ PDRIVER_OBJECT DriverObject,
+    _In_ PUNICODE_STRING RegistryPath
+)
+/*++
+
+Routine Description:
+
+    DriverEntry initializes the driver and is the first routine called by the
+    system after the driver is loaded. DriverEntry specifies the other entry
+    points in the function driver, such as EvtDevice and DriverUnload.
+
+Parameters Description:
+
+    DriverObject - represents the instance of the function driver that is loaded
+    into memory. DriverEntry must initialize members of DriverObject before it
+    returns to the caller. DriverObject is allocated by the system before the
+    driver is loaded, and it is released by the system after the system unloads
+    the function driver from memory.
+
+    RegistryPath - represents the driver specific path in the Registry.
+    The function driver can use the path to store driver related data between
+    reboots. The path does not store hardware instance specific data.
+
+Return Value:
+
+    A success status as determined by NT_SUCCESS macro, if successful.
+
+--*/
```

Comment:
I know other kernel specific code has this, but let's not do it here. It's always been a style diversion from the rest of the code anyways.